### DISCUSSION TITLE
About metis

### DISCUSSION THREAD

**Discussion by zsy0828** [PaddlePaddle/PGL#424] [General]
I also tend to use cython accelerate the graoh training speed. Now I need to use third party metis without add other files, it will output  error LNK1120: 1 unresolved externals.

Your pgl's third party metis just add win.h and win.c which have fake function like METIS_Kway_win32  and it works.

Why it works in cython. I read cython's document in using c++ third party , it hasn't talk about add files such as  win.h.
Does it mean when use third party in cython , we must to add win.h and win.c to make pyx work？I mean use a fake func to compile， and use original func to work, because I only see win == 0.

**Comment by DesmonDay** [comment]
We add fake function for Windows because we had compilation problem at that time. Our metis actually works for Linux and Mac, but not for Windows. Maybe later we will spend some time to solve this problem.

**Reply by zsy0828** [reply]
thank you for your answer.

**Reply by zsy0828** [reply]
> We add fake function for Windows because we had compilation problem at that time. Our metis actually works for Linux and Mac, but not for Windows. Maybe later we will spend some time to solve this problem.

你好，请问pgl的metis和源码除了增加假的win函数外，还有什么区别么，因为在我们的图算法库后续可能也会使用cython来实现metis
